### PR TITLE
WIP example snowplow events on codetime activation

### PR DESCRIPTION
**Pull request by bjacobson26**

### CODE REVIEW

File: src/snowplow/events.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const snowplow = require('snowplow-tracker');
+const emitter = snowplow.emitter;
+const tracker = snowplow.tracker;
+
+// TODO: grab this from server
+const e = emitter("com-software-prod1.mini.snplow.net")
+const t = tracker([e], "CodeTimeTracker", "CodeTime", false);
+
+export function codeEvent(type: string, jwt?: string) {
+	console.log("CODE EVENT", type)
```
Comment (xavluiz) [inline src/snowplow/events.ts:10]: lets use the "logIt" function from the Util.ts as that is using a formatted string

File: src/snowplow/events.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+		"schema": "iglu:com.software/code_event/jsonschema/1-0-0",
+		"data": {
+			"type": type,
+			"jwt": jwt
+		}
+	})
+};
+
+export function editorAction(entity: string, type: string, tz_offset_minutes: number) {
+	console.log("EDITOR ACTION", entity);
```
Comment (xavluiz) [inline src/snowplow/events.ts:21]: same here

File: src/snowplow/events.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+const snowplow = require('snowplow-tracker');
+const emitter = snowplow.emitter;
+const tracker = snowplow.tracker;
+
+// TODO: grab this from server
+const e = emitter("com-software-prod1.mini.snplow.net")
```
Comment (xavluiz) [inline src/snowplow/events.ts:6]: should this get pulled in from api.software.com?

File: src/snowplow/events.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+const t = tracker([e], "CodeTimeTracker", "CodeTime", false);
+
+export function codeEvent(type: string, jwt?: string) {
+	console.log("CODE EVENT", type)
+	t.trackUnstructEvent({
+		"schema": "iglu:com.software/code_event/jsonschema/1-0-0",
```
Comment (xavluiz) [inline src/snowplow/events.ts:12]: also, will we fetch any schema or hard-coded info from our api as other editors will also need to implement something like this

File: src/extension.ts [hunk 5]
```diff
@@ -109,6 +110,9 @@ export function deactivate(ctx: ExtensionContext) {
 }
 
 export async function activate(ctx: ExtensionContext) {
+    // test snowplow event
+    codeEvent("codetime activated")
+    editorAction("editor", "unfocus", -420)
```
Comment (brettmstevens7) [inline src/extension.ts:115]: Our TZ offsets follow Python standard, so this would be 420 for PST

File: src/snowplow/events.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+			"type": type,
+			"jwt": jwt
+		}
+	})
+};
+
+export function editorAction(entity: string, type: string, tz_offset_minutes: number) {
+	console.log("EDITOR ACTION", entity);
+	t.trackUnstructEvent({
+		"schema": "iglu:com.software/editor_action/jsonschema/1-0-0",
```
Comment (xavluiz) [inline src/snowplow/events.ts:23]: same here. seems like we need a general manager to pull in config info and store it when its needed

**Comment by xavluiz** [comment]
general comments. lets discuss tomorrow about what we can do about not hard-coding anything.

File: src/extension.ts (hunk 5 above)
Comment (pbstriker38) [reply src/extension.ts:115]: PDT is minus 7

**Pull request closed without merging**